scripts/pod_lib_lint.sh

```diff
@@ -28,30 +28,38 @@ podspec is the podspec to lint
 
 options can be any options for pod spec lint
 
-Optionally, ALT_SOURCES can be set in the script to add extra repos to the
-search path. This is useful when APIs to core dependencies like FirebaseCore or
-GoogleUtilities and the public podspecs are not yet updated.
 EOF
   exit 1
 fi
 
-# Set ALT_SOURCES like the following to continue lint testing until release when Utilities
-# or Core APIs change. GoogleUtilities.podspec and FirebaseCore.podspec should be
-# manually pushed to a temporary Specs repo. See
-# https://guides.cocoapods.org/making/private-cocoapods.
-ALT_SOURCES="--sources=https://github.com/Firebase/SpecsStaging.git,https://github.com/CocoaPods/Specs.git"
-
 command=(bundle exec pod lib lint "$@")
 
-if [[ -n "${ALT_SOURCES:-}" ]]; then
-  command+=("${ALT_SOURCES}")
+# CocoaPods 1.7.0 added the `--include-podspecs` option for `pod lib lint` to
+# find dependencies locally instead of via a specs repo. This enables testing
+# changes across multiple pods without needing to push the dependent pods to
+# a staging repo.
+
+if ! [[ $1 == "GoogleUtilities.podspec" ||
```

Comment:
This duplicates the information already in the podspec. I'm willing to bet it's less code to just interrogate the podspec. One sec and let me cook up an example.

Reply:
This should do it:

```
#!/usr/bin/env ruby

require 'cocoapods'

def usage()
  script = File.basename($0)
  STDERR.puts <<~EOF
  USAGE: #{script} podspec
  EOF
end

def main(args)
  if args.size < 1 then
    usage()
    exit(1)
  end

  find_local_deps(args[0])
end

def find_local_deps(podspec_file)
  result = []

  spec_dir = File.dirname(podspec_file)

  spec = Pod::Spec.from_file(podspec_file)
  spec.dependencies.each do |dep|
    dep_file = File.join(spec_dir, "#{dep.name}.podspec")
    if File.exist?(dep_file) then
      result << File.basename(dep_file)
    end
  end

  puts result.join(',')
end

main(ARGV)
```

Reply:
Or does this need to be recursive?

Reply:
Yep. It needs to be recursive through continuous locally found podspecs